src/android/jni/debug_protocol.cpp

```diff
@@ -2,152 +2,61 @@
 // Copyright(c) 2019 Intel Corporation. All Rights Reserved.
 
 #include <jni.h>
-#include <memory>
-#include <vector>
 #include "error.h"
-#include <iostream>
 #include <fstream>
-#include <string>
+#include "../../../include/librealsense2/h/rs_internal.h"
+#include "../../../common/parser.hpp"   //needed for auto completion in nGetCommands
 
-#include "../../../include/librealsense2/rs.h"
-#include "../../../include/librealsense2/hpp/rs_device.hpp"
-#include "../../api.h"
-#include "../../../common/parser.hpp"
 
-using std::string;
-using std::hex;
-
-std::vector<uint8_t> send_and_receive_raw_data(JNIEnv *env, rs2_device * dev, const std::vector<uint8_t>& input)
-{
-    std::vector<uint8_t> results;
+extern "C"
+JNIEXPORT jbyteArray JNICALL
+Java_com_intel_realsense_librealsense_DebugProtocol_nSendAndReceiveRawData(JNIEnv *env, jclass type,
```

Comment:
Do we have to use such long names ? We try to API  function names in wrappers as close to the origin as possible

Reply:
This is the way JNI methods names must be used.
The pattern is "Java"_<package_name>_<Java_class_name_in_which_the_jni_method_is_declared>_<method_name>.
Without this pattern, the JNI method is not recognized by the compiler.